Source/Session/VisitOptionsHandler.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+class VisitOptionsHandler {
+    
+    private var unhandledVisitOptions: VisitOptions?
+    
+    /// If a form submission provides a response HTML, save the options and pass them to the next visit proposal.
+    func process(_ options: VisitOptions?) -> VisitOptions {
+        
+        if let options, options.response != nil {
+            /// Options are provided for the next visit.
+            unhandledVisitOptions = options
+            return options
+        } else if let unhandledVisitOptions {
+            /// Next visit is happening. Use the previous visit options.
+            self.unhandledVisitOptions = nil
```

Comment:
Make sure URLs match between visit options

Reply:
Has this been resolved?

Reply:
Not exactly. Would we keep the visit options around indefinitely until the next visit or should the next visit use them? If the former, URL matching makes sense. Otherwise, this implementation works for our purposes.

What do you think? Sorry to say I'm still trying to understand when to expect a `responseHTML` and I also don't think Android is doing it 🤔.